src/Promitor.Agents.ResourceDiscovery/Graph/AzureResourceGraph.cs

```diff
@@ -105,8 +140,20 @@ private async Task<TResponse> InteractWithAzureResourceGraphAsync<TResponse>(str
                         {
                             if (responseException.Response.StatusCode == HttpStatusCode.Forbidden)
                             {
-                                var unauthorizedException = new UnauthorizedException(QueryApplicationId, targetSubscriptions);
-                                _logger.LogCritical(unauthorizedException, "Unable to query Azure Resource Graph");
+                                UnauthorizedException unauthorizedException;
```

Comment:
Maybe we should add the type of auth to the exception?

Reply:
Indeed ! :)